src/AElf.Kernel.SmartContractExecution/Application/IBlockAttachService.cs

```diff
@@ -9,6 +9,7 @@ namespace AElf.Kernel.SmartContractExecution.Application
     public interface IBlockAttachService
     {
         Task AttachBlockAsync(Block block);
+        Task AttachReceivedBlock(BlockWithTransactions blockWithTransactions);
```

Comment:
When should AttachReceivedBlock be used and when should AttachBlockAsync be used?
It seems AttachBlockAsync is only used after mining, then it should not include the logics for executing again.

Reply:
Correct, AttachReceivedBlock is used for blocks received from the network and AttachBlockAsync is intended for mined blocks.

This is an interesting comment, I largely reused the behavior from dev. Currently, on dev, the mined block will be executed twice. I'm not sure we can remove without refactoring because the _AttachBlockToChainAsync_ method will modify the state of the _ChainBlockLinks_ and this is further updated by executing. If you simply remove execution, it will leave the node in an incorrect state after mining.

In the end it seems that the operations after mining can be simplified.